network/discovery/dv5_filters.go

```diff
@@ -13,23 +13,25 @@ func (dvs *DiscV5Service) limitNodeFilter(node *enode.Node) bool {
 }
 
 //// forkVersionFilter checks if the node has the same fork version
-// func (dvs *DiscV5Service) forkVersionFilter(node *enode.Node) bool {
+// func (dvs *DiscV5Service) forkVersionFilter(logger *zap.Logger, node *enode.Node) bool {
 //	forkv, err := records.GetForkVersionEntry(node.Record())
 //	if err != nil {
-//		dvs.logger.Warn("could not read fork version from node record", zap.Error(err))
+//		logger.Warn("could not read fork version from node record", zap.Error(err))
 //		return false
 //	}
 //	return dvs.forkv == forkv
 //}
 
 // badNodeFilter checks if the node was pruned or have a bad score
-func (dvs *DiscV5Service) badNodeFilter(node *enode.Node) bool {
-	pid, err := PeerID(node)
-	if err != nil {
-		dvs.logger.Warn("could not get peer ID from node record", zap.Error(err))
-		return false
+func (dvs *DiscV5Service) badNodeFilter(logger *zap.Logger) func(node *enode.Node) bool {
+	return func(node *enode.Node) bool {
+		pid, err := PeerID(node)
+		if err != nil {
+			logger.Warn("could not get peer ID from node record", zap.Error(err))
+			return false
+		}
+		return !dvs.conns.IsBad(logger, pid)
 	}
-	return !dvs.conns.IsBad(pid)
 }
```

Comment:
why was it necessary to wrap the func and not just add a new arg?

Reply:
because It will not accept filter interface (which is actually `func(node *enode.Node) bool`)